### PR TITLE
feat(accordion): change buttonWidth prop type to string (FE-5709)

### PR DESCRIPTION
**Pull request by grabkowski**
addresses #5897

### Proposed behaviour

buttonWidth prop should accept all valid string CSS width values

### Current behaviour

buttonWidth prop accepts only numbers and a "px" string is added in CSS

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
~~- [ ] Screenshots are included in the PR if useful~~
~~- [ ] All themes are supported if required~~
~~- [ ] Unit tests added or updated if required~~
~~- [ ] Cypress automation tests added or updated if required~~
~~- [ ] Storybook added or updated if required~~
~~- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required~~
- [x] Typescript `d.ts` file added or updated if required
~~- [ ] Related docs have been updated if required~~

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/components/accordion/accordion.component.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ export interface AccordionProps
   extends StyledAccordionContainerProps,
     SpaceProps {
   /** Width of the buttonHeading when it's set, defaults to 150px */
-  buttonWidth?: number;
+  buttonWidth?: string;
```
Comment (edleeks87) [inline src/components/accordion/accordion.component.tsx:24]: I'm wondering if we should avoid the breaking change here and support both `number` and `string`. If it's a number value then we can keep the current behaviour of appending `px` to it. If it's a string we just apply the value and let the implementing developers set the unit etc. What do you think?

**Review by edleeks87 — COMMENTED** [review]
Overall the proposed changes look good to me, just the one question around whether we should avoid the breaking changes

File: src/components/accordion/accordion.test.js [hunk 2]
```diff
@@ -300,8 +300,8 @@ context("Testing Accordion component", () => {
         .and("have.attr", "type", "info");
     });
 
-    it.each([[100], [200], [300]])(
-      "should check accordion heading is a button with width %spx",
+    it.each([["100px"], ["200px"], ["300px"]])(
```
Comment (Parsium) [inline src/components/accordion/accordion.test.js:303]:
```suggestion
    it.each(["100px", "200px", "300px"])(
```

**Review by Parsium — COMMENTED** [review]
Just one minor comment from me, happy to approve though once @edleeks87's comment has been addressed 😄

File: src/components/accordion/accordion.spec.tsx [hunk 3]
```diff
@@ -530,21 +530,43 @@ describe("Accordion", () => {
       }
     );
 
-    it.each(Array.from({ length: 10 }).map((_, i) => 100 + i))(
-      "sets the width of the button to the value passed in via the buttonWidth prop",
-      (buttonWidth) => {
+    describe("sets the width of the button to the value passed in via the buttonWidth prop", () => {
+      it.each(["140px", "20%", "0.2"])("when passed as a %s string", () => {
+        const mockButtonWidth = "140px";
```
Comment (Parsium) [inline src/components/accordion/accordion.spec.tsx:535]: I think we've accidentally missed a test parameter here :)
```suggestion
      it.each(["140px", "20%", "0.2"])("when passed as a %s string", (mockButtonWidth) => {
```

**Review by edleeks87 — APPROVED** [review]

**Review by Parsium — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**